packages/react-core/src/components/Slider/Slider.tsx

```diff
@@ -196,26 +207,33 @@ export const Slider: React.FunctionComponent<SliderProps> = ({
     const newPercentage = getPercentage(newPosition, end);
 
     thumbRef.current.style.setProperty('--pf-c-slider--value', `${newPercentage}%`);
-    setValue(newPercentage);
-
-    /* If discrete, snap to closest step value */
-    if (isDiscrete && steps) {
-      const stepIndex = steps.findIndex(step => step.value >= newPercentage);
-      if (steps[stepIndex].value === newPercentage) {
-        snapValue = steps[stepIndex].value;
+    // convert percentage to value
+    const newValue = (newPercentage * (max - min)) / 100 + min;
+    setValue(newValue);
+
+    /* If custom steps are discrete, snap to closest step value */
+    if (!areCustomStepsContinuous && customSteps) {
+      const stepIndex = customSteps.findIndex(stepObj => stepObj.value >= newPercentage);
+      if (customSteps[stepIndex].value === newPercentage) {
+        snapValue = customSteps[stepIndex].value;
       } else {
-        const midpoint = (steps[stepIndex].value + steps[stepIndex - 1].value) / 2;
+        const midpoint = (customSteps[stepIndex].value + customSteps[stepIndex - 1].value) / 2;
         if (midpoint > newPercentage) {
-          snapValue = steps[stepIndex - 1].value;
+          snapValue = customSteps[stepIndex - 1].value;
         } else {
-          snapValue = steps[stepIndex].value;
+          snapValue = customSteps[stepIndex].value;
         }
       }
+    } else if (!customSteps && step !== 1) {
+      // snap to new value if not custom steps and step is not 1
+      snapValue = Math.round((newValue - min) / step) * step + min;
     }
 
-    // Call value change callback
-    if (onValueChange && !isDiscrete && !steps) {
-      onValueChange(newPercentage);
+    // Call value change callback unless custom steps are discrete or no custom steps and step is 1.  The value will snap after the thumb is released.
+    if (onChange) {
+      if ((areCustomStepsContinuous && customSteps) || (!customSteps && step === 1)) {
```

Comment:
Now the `onChange` is only being called if `step={1}` or the prop is omitted and it defaults to 1, so in the examples the values do not update until the mouse is released if the steps are anything above 1 (the opposite issue of what it was previously).  Is there a reason to limit `onChange` from firing based on the step?  It seems like that would prevent the user from seeing their current value without releasing the mouse.

Reply:
Hmmmm. I think I had issue with that.  I think it was more with the interaction of dragging the thumb not looking smooth  if you update the value and snap before releasing the thumb.  That is just my opinion though.  I can update it and see which interaction is better.  I will keep it the same for custom steps so we can see the difference.

Reply:
Or would you still not expect the slider thumb to snap until after release?
cc @mcarrano

Reply:
I'm not sure what the issue here is @tlabaj . It's currently working as I'd expect it to. We're talking about this example, right? https://patternfly-react-pr-5669.surge.sh/components/slider#thumb-value-input

Reply:
@mcarrano @evwilkin, I updated the slider to change value as you drag for non custom steps as seen in the second and third discrete examples.  The first discrete example with custom steps snaps and updates the value once the thumb is released.  Which is the correct and logic interaction?

Reply:
OK. Now I better understand the question @tlabaj . I think we should go with the implementation in the first example - update only after the thumb is released. If I understand correctly, this is also how the standard HTML range input works? I tested this on my phone also and it seems to work well there.

FWIW, I initially thought that the second example was broken because the steps are far apart and the slider does not move until you get close enough to the next discrete value. I doubt that users would understand that.

Reply:
So the standard html range actual updates as you drag it. Like in the second example (seen in first clip).   Heres is a recording of HTML example.  This make sense like Evan said, especially if you are displaying the value as you drag the thumb.  

https://user-images.githubusercontent.com/39532947/115035338-7d57f280-9e9a-11eb-8e34-e45b06445462.mov

Reply:
OK. I misunderstood. Then yes, I think we should follow what the standard HTML slider does. However, I would then either change the second example to have tick marks or remove it entirely. From a design perspective I would not recommend not using tick marks when the slider has discrete steps. As a user I have no way of telling what the increments are without them.